File: Kotlin/src/Kotlin/map/map.kt
package Kotlin.map
import Kotlin.Item
import java.time.LocalDateTime

fun main(){
	var items:Map<String,Item> = mapOf<String,Item>(
		"Dektop" to Item(1,"Item1",LocalDateTime.of(2021,2,25,5,32,50)),
		"Laptop" to Item(2,"Item2",LocalDateTime.of(2021,3,25,6,3,50)),
		"Mobile" to Item(1,"Item1",LocalDateTime.of(2020,8,20,8,45,30)),
		"SmartWatch" to Item(1,"Item1",LocalDateTime.of(2029,4,3,12,23,43))
	)
	
	items.forEach{k:String,v:Item->
	print(v)//print(items[k])
	}
	println()
	items += "New" to Item(1,"Item1",LocalDateTime.of(2029,4,3,12,23,43))
	items.toSortedMap(compareBy<String>{it}.thenBy{items[it]?.id})
}

File: dart/map/map.dart
import '../Item.dart';

void main() {
  Map<String, Item> items = {
    'Laptop': Item('Item1', 4, DateTime(2021, 5, 25, 6, 45, 56)),
    '''Desktop''': Item('Item2', 2, DateTime(2021, 6, 25, 6, 45, 56)),
    r'Mobile': Item('Item3', 1, DateTime(2029, 5, 25, 6, 45, 56)),
    "Tablet": Item('Item4', 3, DateTime(2020, 5, 25, 6, 45, 56))
  };
  //adding
  items["New"] = Item('Item5', 5, DateTime(2020, 6, 5, 6, 45, 56));
  items.remove("New"); //remove by key
  //traversal
  items.forEach((String key, Item value) {
    print(value.toString());
  });
  //sort keys
  items.keys.toList().sort();
}
